src/ui/public/chrome/api/injected_vars.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { cloneDeep } from 'lodash';
+import { InjectedMetadataStartContract } from '../../../../core/public/injected_metadata';
+
+let newPlatformInjectedVars: InjectedMetadataStartContract;
+
+export function __newPlatformInit__(instance: InjectedMetadataStartContract) {
+  if (newPlatformInjectedVars) {
+    throw new Error('ui/chrome/api/injected_vars is already initialized');
+  }
+
+  newPlatformInjectedVars = instance;
+}
+
+export function initChromeInjectedVarsApi(chrome: any) {
+  chrome.getInjected = (name: string, defaultValue: any) =>
+    cloneDeep(
```

Comment:
question: Why cloneDeep here? In the original implementation, it was only `clone` and it only applied when dealing with all vars rather than the case where you were retrieving a single var.

Reply:
I can't think of a problem it would cause. I just noticed it was different and I know we're trying to preserve behaviors where we can during the migration to the new platform.